python/distrdf/dask/test_backend.py

```diff
@@ -0,0 +1,245 @@
+        # the number of spawned workers.
+        df = Dask.RDataFrame(1, daskclient=self.client).Define("u", "userValue").Histo1D("u")
+        h = df.GetValue()
+        self.assertEqual(h.GetMean(), 123)
+
+
+class EmptyTreeErrorTest(unittest.TestCase):
+    """
+    Distributed execution fails when the tree has no entries.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up test environment for this class. Currently this includes:
+
+        - Initialize a Dask client for the tests in this class. This uses a
+          `LocalCluster` object that spawns 2 single-threaded Python processes.
+        """
+        cls.client = Client(LocalCluster(n_workers=2, threads_per_worker=1, processes=True))
+
+    @classmethod
+    def tearDownClass(cls):
+        """Reset test environment."""
+        cls.client.shutdown()
+
+    def test_histo_from_empty_root_file(self):
+        """
+        Check that when performing operations with the distributed backend on
+        an RDataFrame without entries, DistRDF raises an error.
+        """
+
+        # Create an RDataFrame from a file with an empty tree
+        rdf = Dask.RDataFrame("NOMINAL", "../emptytree.root", daskclient=self.client)
+        histo = rdf.Histo1D(("empty", "empty", 10, 0, 10), "mybranch")
+
+        # Get entries in the histogram, raises error
+        with self.assertRaises(RuntimeError):
+            histo.GetEntries()
+
+
+class ChangeAttributeTest(unittest.TestCase):
+    """Tests that check correct changes in the class attributes"""
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up test environment for this class. Currently this includes:
+
+        - Initialize a Dask client for the tests in this class. This uses a
+          `LocalCluster` object that spawns 2 single-threaded Python processes.
+        """
+        cls.client = Client(LocalCluster(n_workers=2, threads_per_worker=1, processes=True))
+
+    @classmethod
+    def tearDownClass(cls):
+        """Reset test environment."""
+        cls.client.shutdown()
+
+    def test_change_attribute_when_npartitions_greater_than_clusters(self):
+        """
+        Check that the `npartitions class attribute is changed when it is
+        greater than the number of clusters in the ROOT file.
+        """
+
+        treename = "mytree"
+        filename = "myfile.root"
+        ROOT.RDataFrame(100).Define("x", "rdfentry_").Snapshot(treename, filename)
+
+        df = Dask.RDataFrame(treename, filename, npartitions=10, daskclient=self.client)
+
+        self.assertEqual(df._headnode.npartitions, 10)
+        histo = df.Histo1D("x")
+        nentries = histo.GetEntries()
+
+        self.assertEqual(nentries, 100)
+        self.assertEqual(df._headnode.npartitions, 1)
+
+        os.remove(filename)
+
+    def test_user_supplied_npartitions_have_precedence(self):
+        """
+        Check that even if Dask configuration options could optimize the number
```

Comment:
By `Dask configuration options` here you mean the internal optimization algorithm of the Dask backend for the number of partitions?

Reply:
Mmm yeah sort of. In this case the dask client is connected to a `LocalCluster` with 2 cores, but the user passes `npartitions=4` so it should still stay at 4 even though the backend's `optimize_npartitions` would return 2

Reply:
I'll improve the docs